pypdf/generic/_base.py

```diff
@@ -123,6 +123,7 @@ def _reference_clone(
         if ind is not None:
             if id(ind.pdf) not in pdf_dest._id_translated:
                 pdf_dest._id_translated[id(ind.pdf)] = {}
+                pdf_dest._id_translated[id(ind.pdf)][-10] = ind.pdf
```

Comment:
Why -10? Doesn't this potentially override something important?

Reply:
No : negative ids are not used ; this mod only link the source pdf to the PdfWriter object, preventing its garbage collection and then therefore the id reuse. the GC will be performed when the source object will be reused

Reply:
What does the `-10`mean? I've just noticed that this is a dictionary, not a list. Does -10 have  any special meaning?

Reply:
Nothing special I selected a number far enough to leave immediate negative numbers available for other use